src/providers/azure/azure.go

```diff
@@ -118,18 +132,9 @@ func assertFabricCompatible(endpoint net.IP, desiredVersion string) error {
 }
 
 func fetchSharedConfig(endpoint net.IP) (config.Metadata, error) {
-	client := retry.Client{
-		InitialBackoff: time.Second,
-		MaxBackoff:     time.Second * 5,
-		MaxAttempts:    10,
-		Header: map[string][]string{
-			"x-ms-agent-name": {AgentName},
-			"x-ms-version":    {FabricProtocolVersion},
-			"Content-Type":    {"text/xml; charset=utf-8"},
-		},
-	}
+	client := getClient(true)
```

Comment:
I'd rather you attach the agent to everthing than have a random boolean parameter.